onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_builder.cc

```diff
@@ -397,10 +411,13 @@ static Status AddInitializerTransposed(ModelBuilder& model_builder,
   std::unique_ptr<uint8_t[]> buffer_holder(new uint8_t[operand_type.GetOperandBlobByteSize()]);
   uint8_t* buffer = buffer_holder.get();
   size_t element_size = operand_type.GetElementByteSize();
+  // If necessary, convert int8 tensor to uint8 tensor for per-tensor u8s8
+  // i^0x80 == i + 128
+  uint8_t bit_flip_val = is_per_tensor_u8s8 ? 0x80 : 0;
   for (uint32_t x = 0; x < x_t; x++) {
```

Comment:
nit: is_per_tensor_u8s8 is a bit opaque as you need a lot of implicit knowledge to know that that translates to having to convert the 's8' part to unsigned. 

Also wondering if you need to pass the value in given you know the data type from the Tensor instance.

Reply:
Updated comments,